platforms/pf_test/platform.cpp

```diff
@@ -249,6 +258,87 @@ static void *_task_planner_start_cancel(void *arg)
     return NULL;
 }
 
+/// Execute a shell command callback using arguments from Protobuf message.
+/// Command name is in the 'cmd' attribute of the Protobuf message.
+/// Arguments are in a space-separated string in 'desc' attribute of the Protobuf message.
+static void run_command(cogip::shell::Command *command, const cogip::shell::Command::PB_Message *pb_command)
+{
+    // Computed number of arguments
+    int argc = 0;
+    // List of arguments null-separated
+    char args[COMMAND_DESC_MAX_LENGTH];
+    // Array of pointers to each argument
+    char *argv[MAX_COMMAND_ARGS];
+    // First argument is the command
+    argv[argc++] = (char *)pb_command->cmd();
+    if (pb_command->get_desc().get_length() != 0) {
+        // If there are arguments to the command in the 'desc' attribute
+        size_t i;
+        // Copy first argument pointer to 'argv'
+        argv[argc++] = args;
+        for (i = 0; i < pb_command->get_desc().get_length(); i++) {
+            assert(i < COMMAND_DESC_MAX_LENGTH);
+            assert(argc < MAX_COMMAND_ARGS);
```

Comment:
assert are not a bit overkill in such case ?

Reply:
Yes, now I skip the command if its too long, and print a message.